pylint/extensions/kwargs.py

```diff
@@ -0,0 +1,66 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+# Copyright (c) https://github.com/PyCQA/pylint/blob/main/CONTRIBUTORS.txt
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING
+
+from astroid import BoundMethod, nodes
+
+from pylint.checkers import BaseChecker
+from pylint.checkers.utils import safe_infer
+from pylint.interfaces import INFERENCE
+
+if TYPE_CHECKING:
+    from pylint.lint import PyLinter
+
+
+class KeywordChecker(BaseChecker):
+    """Checks related to keyword arguments."""
+
+    name = "kwargs"
+    msgs = {
+        "W3501": (
+            "Call to `%s` missing keyword argument `%s`.",
+            "missing-keyword-arg",
+            "Used when method is called without specifying a keyword argument. ",
+        ),
+    }
+
+    def visit_call(self, node: nodes.Call) -> None:
+        """Check that called functions/methods are provided keyword arguments."""
+        called = safe_infer(node.func)
+
+        if isinstance(called, nodes.ClassDef):
+            needed_keywords = called.instance_attrs.keys()
+            default_kwargs = []  # todo
```

Comment:
@DanielNoord would you happen to recall how to get the default kwargs for methods like `__init__`? I got it working for normal functions and instance methods but couldn't find it for these. If not, I could make the checker not check stuff like this `car = Car("a car", "name", "title"`.

Reply:
Doesn't `ClassDef.args` have a method for that?`default_value()`.

Reply:
```
(Pdb) called.args
*** AttributeError: 'ClassDef' object has no attribute 'args'
```

?

Reply:
``node.locals["__init__"].args``. Sorry!

Reply:
Awesome, super helpful!

The issue I'm having now is that I don't have a way to connect `defaults` to their name.  For this example

```
def make_car(name, owner, title, color="red", year=2022):
    return "car"

second_car = make_car("a car", "name", "title") # [missing-keyword-arg, missing-keyword-arg, missing-keyword-arg]
```

We only want to raise the new message 3x, not 5x, because color and year are defaults and the user didn't provide them and that's ok. 

So I need to figure out a way to tell my new checker to remove these defaults, but I don't have their names... all I have is

```
(Pdb) called.args.defaults
[<Const.str l.82 at 0x11a4da4f0>, <Const.int l.82 at 0x11a4da610>]
```

```
(Pdb) called.args.args
[<AssignName.name l.82 at 0x11a4da400>, <AssignName.owner l.82 at 0x11a4da520>, <AssignName.title l.82 at 0x11a4da550>, <AssignName.color l.82 at 0x11a4da580>, <AssignName.year l.82 at 0x11a4da5b0>]
```

but there isn't a way to connect them to each other. Does that make sense? If so, do you have any suggestions?

Reply:
`dict(zip(called.args.args[-len(called.args.defaults):], called.args.defaults))` assuming they are properly parallel lists, I think.

Reply:
Good thinking! Just trying out one example it does seem we can assume order